src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -796,6 +878,11 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
   };
 };
 
+void VM_Version::report_apx_state_restore_warning() {
+  tty->print("warning: Unsuccessful EGPRs state restoration across signal handling, setting UseAPX to false.\n");
```

Comment:
This print is fine during development but I would instead save some value in memory to indicate that OS does not save/restore APX. And then check it after we execute this assembler code. Similar how we do that for AVX.
You would not need to do runtime call and this method then.
Note: `tty->print()` can do "nasty"/unexpected things which you want to avoid.

Reply:
Hi @vnkozlov , doing a lazy restored state comparison now to align with existing AVX handling.